### PR TITLE
CORE-1839: Change UserPlan to Subscription

### PR DESCRIPTION
**Pull request by johnworth**
This PR is part of the effort to rename the user_plans table in the QMS database to subscriptions. This includes changing all type names, references to user_plans in the SQL, and all NATS subjects containing references to user_plans or UserPlans.

### CODE REVIEW

File: src/terrain/routes/schemas/qms.clj [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
-(def UpdateUserPlanQuotaSummary "Updates a quota in the user's current subscription")
-(def UpdateUserPlanQuotaDescription "Updates the user's quota for the specified resource type")
-(def UpdateUserPlanSummary "Changes the user's plan to another one in QMS")
-(def UpdateUserPlanDescription "Changes the user's plan to another one in QMS. New plan is referenced by name")
+(def GetSubscriptionSummary "Returns details about the user's current plan")
+(def GetSubscriptionDescription "Returns details about the user's current plan, including quota and usage information")
+(def UpdateSubscriptionQuotaSummary "Updates a quota in the user's current subscription")
+(def UpdateSubscriptionQuotaDescription "Updates the user's quota for the specified resource type")
+(def UpdateSubscriptionSummary "Changes the user's plan to another one in QMS")
+(def UpdateSubscriptionDescription "Changes the user's plan to another one in QMS. New plan is referenced by name")
```
Comment (slr71) [inline src/terrain/routes/schemas/qms.clj:17]: These descriptions could be reworded a bit to reflect the change as well.

File: src/terrain/routes/schemas/resource_usage_api.clj [hunk 2]
```diff
@@ -125,10 +125,10 @@
    :users                (describe QMSUser "User information from the quota management system")})
 
 (defschema ResourceSummary
-  {:cpu_usage  (describe CPUHoursTotal "The object containing the CPU hours total")
-   :data_usage (describe UserCurrentDataTotal "The object containing the data usage total")
-   :user_plan  (describe UserPlan "The user's current plan")
-   :errors     (describe [APIError] "The list of errors generated during summary creation")})
+  {:cpu_usage     (describe CPUHoursTotal "The object containing the CPU hours total")
+   :data_usage    (describe UserCurrentDataTotal "The object containing the data usage total")
+   :subscription  (describe Subscription "The user's current plan")
```
Comment (slr71) [inline src/terrain/routes/schemas/resource_usage_api.clj:130]: This description could be reworded a bit to reflect the change.

**Review by slr71 — APPROVED** [review]
I had a couple of minor suggestions, but I didn't catch anything that would cause a problem :shipit:

File: src/terrain/routes/schemas/qms.clj (hunk 1 above)
Comment (johnworth) [reply src/terrain/routes/schemas/qms.clj:17]: This should be fixed now.

**Pull request merged**